### PR TITLE
ST: Improve log collecting in case of test failure

### PR DESCRIPTION
**Pull request by Frawless**
### Type of change

- Bugfix
- Refactoring

### Description

This PR improves the way how the logs from pods and events are stored in case of test failure. Changes there also changes a structure a little bit:
```
test-class:
   test-method-0:
       namespace-0
       namespace-1
   test-method-1:
       namespace-2
       namespace-3
   co-logs (shared)
```

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java [hunk 1]
```diff
@@ -99,106 +101,110 @@ public LogCollector(String testSuiteName, String testCaseName, KubeClient kubeCl
...
+            namespaces.add(clusterOperatorNamespace);
             // collect logs for all namespace related to test suite
-            namespaces.stream().forEach(namespace -> {
-                namespaceFile = new File(testCase + "/" + namespace);
+            namespaces.forEach(namespace -> {
+                if (collectorElement.get().isEmpty()) {
+                    namespaceFile = new File(this.testCase + "/" + namespace);
+                } else {
+                    namespaceFile = new File(this.testSuite +  "/" + namespace);
+                }
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java:121]: Would it make sense to use ternary operator here?

File: systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java [hunk 2]
```diff
@@ -24,7 +25,9 @@ public void handleTestExecutionException(ExtensionContext extensionContext, Thro
         if (!(throwable instanceof TestAbortedException)) {
             final String testClass = extensionContext.getRequiredTestClass().getName();
             final String testMethod = extensionContext.getRequiredTestMethod().getName();
-            collectLogs(testClass, testMethod);
+            CollectorElement collectorElement = new CollectorElement(testClass, testMethod);
+
+            collectLogs(collectorElement);
```
Comment (im-konge) [inline systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java:30]: How about
```
collectLogs(new CollectorElement(testClass, testMethod));
```
?

**Review by im-konge — APPROVED** [review]
Just two nits, LGTM otherwise

File: systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java (hunk 1 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java:121]: This is more readable.

File: systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java (hunk 2 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/logs/TestExecutionWatcher.java]: Updated

File: systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/main/java/io/strimzi/systemtest/logs/LogCollector.java:121]: Yes, make sense.

**Review by scholzj — APPROVED** [review]

**Pull request merged**